test/e2e/predictor/test_lightgbm.py

```diff
@@ -125,12 +122,115 @@ def test_lightgbm_v2_runtime_kserve():
+                requests={"cpu": "50m", "memory": "128Mi"},
+                limits={"cpu": "1", "memory": "1Gi"},
+            ),
+        ),
+    )
+
+    isvc = V1beta1InferenceService(
+        api_version=constants.KSERVE_V1BETA1,
+        kind=constants.KSERVE_KIND,
+        metadata=client.V1ObjectMeta(
+            name=service_name, namespace=KSERVE_TEST_NAMESPACE
+        ),
+        spec=V1beta1InferenceServiceSpec(predictor=predictor),
+    )
+
+    kserve_client = KServeClient(
+        config_file=os.environ.get("KUBECONFIG", "~/.kube/config"))
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(
+        service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    res = predict(service_name, "./data/iris_input_v2.json",
+                  protocol_version="v2")
+    assert res["outputs"][0]["data"] == [
+        8.796664107010673e-06,
+        0.9992300031041593,
+        0.0007612002317336916,
+        4.974786820804187e-06,
+        0.9999919650711493,
+        3.0601420299625077e-06]
+
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+@pytest.mark.grpc
+def test_lightgbm_v2_grpc():
+    service_name = "isvc-lightgbm-v2-grpc"
+    model_name = "lightgbm"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        model=V1beta1ModelSpec(
+            model_format=V1beta1ModelFormat(
+                name="lightgbm",
+            ),
+            runtime="kserve-lgbserver",
+            storage_uri="gs://kfserving-examples/models/lightgbm/v2/iris",
+            resources=V1ResourceRequirements(
+                requests={"cpu": "50m", "memory": "128Mi"},
+                limits={"cpu": "1", "memory": "1Gi"},
+            ),
+            ports=[
+                V1ContainerPort(
+                    container_port=8081,
+                    name="h2c",
+                    protocol="TCP"
+                )],
+            args=["--model_name", model_name, "--protocol", "grpc-v2"]
+        )
+    )
+
+    isvc = V1beta1InferenceService(api_version=constants.KSERVE_V1BETA1,
+                                   kind=constants.KSERVE_KIND,
+                                   metadata=client.V1ObjectMeta(
+                                       name=service_name, namespace=KSERVE_TEST_NAMESPACE),
+                                   spec=V1beta1InferenceServiceSpec(predictor=predictor))
+
     kserve_client = KServeClient(config_file=os.environ.get("KUBECONFIG", "~/.kube/config"))
     kserve_client.create(isvc)
     kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
 
-    res = predict(service_name, "./data/iris_input_v2.json", protocol_version="v2")
-    assert res["outputs"][0]["data"] == [
+    json_file = open("./data/iris_input_v2_grpc.json")
+    payload = json.load(json_file)["inputs"]
+
+    response = predict_grpc(service_name=service_name, payload=payload, model_name=model_name)
+    fields = response.outputs[0].contents.ListFields()
```

Comment:
I think you should be able to directly access the float content field instead of using general `ListFields` functions